### PR TITLE
Update doc with workspace

### PR DESCRIPTION
**Pull request by MalloZup**
https://github.com/MalloZup/ha-sap-terraform-deployments/blob/doc-work/doc/workspaces-workflow.md

### CODE REVIEW

**Comment by juadk** [comment]
> The unique thing that I would actually comment is that the workspace impacts in the name of the resources as this would become a prefix

Yes that's why we got a failure in AWS and Azure using _ in workspace name (#169).

**Comment by MalloZup** [comment]
yes that's why I have added the note. So can we merge this :grapes:

File: doc/workspaces-workflow.md [hunk 1]
```diff
@@ -14,17 +14,14 @@ Some excerpts:
 
 > Workspaces are technically equivalent to renaming your state file. They aren't any more complex than that. Terraform wraps this simple notion with a set of protections and support for remote state.
 
-The proposed workflow is the following:
 
-  - Workspaces created by any user must be prefixed with `$USER-` and end with a simple but descriptive name for the deployment (e.g., `$USER-testing-ha`) or the GIT branch (e.g., `$USER-add-ha`).  In the case of OpenQA workers, it could be `openqa-$TEST_NUMBER`.
-
-  - Where possible, the tags and labels of the resources created must be prefixed by the workspace, accessible with `${terraform.workspace}` to make it easier to remove resources from the web console when `terraform destroy` can not. Other resources may need to be renamed to conform to this nomenclature to avoid namespace clashes.
```
Comment (ldevulder) [inline doc/workspaces-workflow.md]: Why did you removed this part? FMPOV could be still useful.

Comment (MalloZup) [reply doc/workspaces-workflow.md]: @ldevulder  I removded because it contains the `-` chars.. :grin: 
Let me re-adapt it

File: doc/workspaces-workflow.md [hunk 2]
```diff
@@ -14,17 +14,14 @@ Some excerpts:
...
+```IMPORTANT```: the terraform workspace name **must not contain** `-` or `_` characters.  Otherwise you will encounter failures by different cloud providers
```
Comment (alvarocarvajald) [inline doc/workspaces-workflow.md]: I think `-` is accepted, at least I've used workspace names such as `build-1-35` in Azure the past. Problem with `_` is that it is a character that the DNS RFC forbids as part of the FQDN, and this configuration adds the workspace as a prefix to several resources, including hostnames. Additionally, workspace names which contain `.` would also not be a good idea

File: doc/workspaces-workflow.md [hunk 3]
```diff
@@ -14,17 +14,14 @@ Some excerpts:
...
 
 ## TL;DR
 
 To create a new workspace:
 
-`terraform worspace new $USER-test-amazing-stuff`
+`terraform worspace new $USER`
```
Comment (alvarocarvajald) [inline doc/workspaces-workflow.md]: Same as above. AFAIK, dashes `-` are OK.

File: doc/workspaces-workflow.md [hunk 4]
```diff
@@ -40,7 +37,7 @@ To list all workspaces:
 
 To remove the previously created workspace:
 
-`terraform workspace delete $USER-test-amazing-stuff`
+`terraform workspace delete $USER`
```
Comment (alvarocarvajald) [inline doc/workspaces-workflow.md]: Idem.

**Comment by MalloZup** [comment]
Ok  closing fo moment. thx for review
